docs/src/test/java/sample/javadsl/ConsumerExample.java

```diff
@@ -388,4 +386,60 @@ public void demo() {
   }
 }
 
+// Shutdown via Consumer.Control
+class ShutdownPlainSourceExample extends ConsumerExample {
+  public static void main(String[] args) {
+    new ExternalOffsetStorageExample().demo();
+  }
+
+  public void demo() {
+    // #shutdownPlainSource
+    final DB db = new DB();
+
+    db.loadOffset().thenAccept(fromOffset -> {
+      Consumer.Control control = Consumer.plainSource(
+          consumerSettings,
+          Subscriptions.assignmentWithOffset(new TopicPartition("topic1", 0), fromOffset)
+      ).mapAsync(1, record -> db.save(record))
+          .toMat(Sink.ignore(), Keep.left())
+          .run(materializer);
+
+      // Shutdown the consumer when desired
+      control.shutdown();
+    });
+    // #shutdownPlainSource
+  }
+}
+
+// Shutdown when batching commits
+class ShutdownCommittableSourceExample extends ConsumerExample {
+  public static void main(String[] args) {
+    new AtLeastOnceExample().demo();
+  }
+
+  public void demo() {
+    // #shutdownCommitableSource
+    final DB db = new DB();
+
+    Pair<Consumer.Control, CompletionStage<Done>> r =
+        Consumer.committableSource(consumerSettings, Subscriptions.topics("topic1"))
+            .mapAsync(1, msg ->
+                db.update(msg.record().value()).thenApply(done -> msg.committableOffset()))
+            .batch(20,
+                first -> ConsumerMessage.emptyCommittableOffsetBatch().updated(first),
+                (batch, elem) -> batch.updated(elem))
+            .mapAsync(3, c -> c.commitJavadsl())
+            .toMat(Sink.ignore(), Keep.both())
+            .run(materializer);
+
+    Consumer.Control control = r.first();
+    CompletionStage<Done> done = r.second();
```

Comment:
I wonder if there is a better variable name than `done`, which would make the `thenCompose` read more natural.

Reply:
`streamComplete` would be better. Or do you have something else in mind?

Reply:
Yes, I like that.